govcd/openapi.go

```diff
@@ -351,7 +351,7 @@ func (client *Client) OpenApiPutItemSync(apiVersion string, urlRef *url.URL, par
 //
 // Note. Even though it may return error if the item does not support asynchronous request - the object may still be
 // created. OpenApiPutItem would handle both cases and always return created item.
-func (client *Client) OpenApiPutItemAsync(apiVersion string, urlRef *url.URL, params url.Values, payload interface{}) (Task, error) {
+func (client *Client) OpenApiPutItemAsync(apiVersion string, urlRef *url.URL, params url.Values, payload interface{}, additionalHeader map[string]string) (Task, error) {
```

Comment:
it's bad practice to pass more than 3 parameters in a function, may be use a struct?

Reply:
This is a draft PR. Please refrain from commenting until it is open

Reply:
this pr merged in master, lol

Reply:
@crashiura,
Two things here:
* This PR went into review later on therefore this was valid comment
* We discussed this extensively and decided not to use a `struct` as parameter because it introduces other problems - becomes a black hole of paramters as time goes on. It is better to have longer and explicitly clear parameter than a `struct` which grows over time - it is just masking parameters.